src/inventorymanager.cpp

```diff
@@ -352,10 +353,14 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 	bool allow_swap = !list_to->itemFits(to_i, src_item, &restitem)
 		&& restitem.count == src_item.count
 		&& !caused_by_move_somewhere;
+	int expected_move_count = src_item.count - restitem.count;
```

Comment:
How about updating `count` or `src_item` here? 
Normal stack movement: If a swap is necessary, those two variables will be overwritten anyway 
Move somewhere (Shift-Click): no swapping at all, and the loop earlier in this function ensures that the entire stack is moved.

Reply:
Updated `move_count` instead.